fapolicy_analyzer/ui/faprofiler.py

```diff
@@ -0,0 +1,273 @@
+        self.name = os.path.basename(self.execPath)
+        self.timeStamp = None
+        self.status = ProfSessionStatus.QUEUED
+        self.procTarget = None
+
+        # Temp demo workaround for setting target log file paths
+        if os.environ.get("FAPD_LOGPATH"):
+            self.tgtStdout = os.environ.get("FAPD_LOGPATH") + f"_{self.name}.stdout"
+            self.tgtStderr = os.environ.get("FAPD_LOGPATH") + f"_{self.name}.stderr"
+        else:
+            self.tgtStdout = None
+            self.tgtStderr = None
+
+        # File descriptors are state variables; closed in stopTarget()
+        self.fdTgtStdout = None
+        self.fdTgtStderr = None
+
+    def startTarget(self, instance_count, block_until_termination=True):
+        logging.debug("FaProfSession::startTarget()")
+
+        if not self.tgtStdout:
+            self.timeStamp = self.get_profiling_timestamp()
+            full_basename = f"/tmp/tgt_profiling_{self.timeStamp}_{self.name}"
+            self.tgtStdout = f"{full_basename}_{instance_count}.stdout"
+            self.tgtStderr = f"{full_basename}_{instance_count}.stderr"
+
+        try:
+            self.fdTgtStdout = open(self.tgtStdout, "w")
+            self.fdTgtStderr = open(self.tgtStderr, "w")
+        except Exception as e:
+            print(f"Profiling target redirection failure: {e}")
+            self.fdTgtStdout = None
+            self.fdTgtStderr = None
+
+        # Set pwd - Use current dir for pwd if not supplied by user
+        if self.pwd:
+            working_dir = self.pwd
+        else:
+            working_dir = os.getcwd()
+        logging.debug(f"Profiling tgt pwd: {working_dir}")
+
+        # Convert username to uid/gid
+        u_valid = False
+        try:
+            if self.user:
+                # Get uid/gid of new user and the associated default group
+                pw_record = pwd.getpwnam(self.user)
+                uid = pw_record.pw_uid
+                gid = pw_record.pw_gid
+                logging.debug(f"The uid/gid of the profiling tgt: {uid}/{gid}")
+
+                # Change the ownership of profiling tgt's stdout and stderr
+                logging.debug(f"Changing ownership of fds: {self.fdTgtStdout},{self.fdTgtStderr}")
+                os.fchown(self.fdTgtStdout.fileno(), uid, gid)
+                os.fchown(self.fdTgtStderr.fileno(), uid, gid)
+                logging.debug(f"Changed the profiling tgt stdout/err ownership {uid},{gid}")
+                u_valid = True
+            else:
+                # In production, the following will be the superuser defaults
+                uid = os.getuid()
+                gid = os.getgid()
+
+        except Exception as e:
+            # Use current uid/gid if getpwnam() or chown throws an exception by
+            # setting prexec_fn = None
+            # Typically will only occur in debug/development runs
+
+            print(f"Profiling target effective user change failure: {e}")
```

Comment:
Toaster popup warning?

Reply:
I'll discuss this one w/you further. This code is strictly only to allow the profiling target to run when in a development environment when running as non-root, which is not possible when in a production environment. Other pop-ups (afaik) also had were part of the standard UX.

Reply:
We agreed to go with a `logging.warning()` statement but the more I thought about it, the pop-up is the better approach. My point wrt this failure will only occur in development scenarios was optimistic. The user can enter an effective username that does not exist on the test platform which would generate a runtime exception on a production system.

Add the pop-up. Will push shortly.